fvt/storage/storage_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package storage
+
+package storage
+
+import (
+	. "github.com/kserve/modelmesh-serving/fvt"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var isvcFiles = map[string]string{
+	"isvc-pvc-storage-uri":  "isvc-pvc-uri.yaml",
+	"isvc-pvc-storage-path": "isvc-pvc-path.yaml",
+	"isvc-pvc2":             "isvc-pvc-2.yaml",
+	"isvc-pvc3":             "isvc-pvc-3.yaml",
+	"isvc-pvc4":             "isvc-pvc-4.yaml",
+}
+
+// ISVCs using PVCs from the FVT `storage-config` Secret (config/dependencies/fvt.yaml)
+var isvcWithPvcInStorageConfig = []string{"isvc-pvc-storage-uri", "isvc-pvc-storage-path", "isvc-pvc2"}
+
+// ISVC using PVC not in the FVT `storage-config` Secret (config/dependencies/fvt.yaml)
+// this should work only after setting allowAnyPVC = true
+var isvcWithPvcNotInStorageConfig = "isvc-pvc3"
+
+// ISVC using a PVC that does not exist at all, this ISVC should fail to load
+var isvcWithNonExistentPvc = "isvc-pvc4"
+
+var _ = Describe("ISVCs", Ordered, FlakeAttempts(3), func() {
+
+	Describe("with PVC in storage-config", Ordered, func() {
+
+		for _, name := range isvcWithPvcInStorageConfig {
+
+			Describe("\""+name+"\"", Ordered, func() {
+				var isvcName = name
+				var fileName = isvcFiles[name]
+
+				It("should successfully load a model", func() {
+					isvcObject := NewIsvcForFVT(fileName)
+					isvcName = isvcObject.GetName()
+					CreateIsvcAndWaitAndExpectReady(isvcObject, PredictorTimeout)
+				})
+
+				It("should successfully run inference", func() {
+					ExpectSuccessfulInference_sklearnMnistSvm(isvcName)
+				})
+
+				BeforeAll(func() {
```

Comment:
NIT: to match the order of execution of the code better (and the other test suites), I think `BeforeAll` and `AfterAll` should be above the first `It` in this `Describe`.

Reply:
sure, will move it

Reply:
done